integration/linux/build/Dockerfile-linux.template

```diff
@@ -23,24 +23,27 @@ ENV GCC_VERSION 11
 # rpm on centos 7 iterates over all fds up to the limit, which is
 # extremely slow. Force it to be small with ulimit.
 #
-# We need to disable the mirrorlist both at the start *and* after
-# installing centos-release-scl, which adds new things. Ugh.
-RUN ulimit -n 1024 && echo precedence ::ffff:0:0/96 100 >> /etc/gai.conf  && ( \
-    source /etc/os-release; \
-     [ "$VERSION_ID" = "7" -o "$VERSION_ID" = "8" ] && echo YES && (cd /etc/yum.repos.d/ \
+# We need to disable the mirrorlist and explicitly set up a
+# baseurl. Switch to the archive.kernel.org mirror while we are at it.
+RUN ulimit -n 1024 \
     && sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* \
-    && sed -i 's|# \?baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* \
-    && yum update -y) || true)
+    && sed -i 's|# \?baseurl=http://mirror.centos.org|baseurl=https://archive.kernel.org/centos-vault/|g' /etc/yum.repos.d/CentOS-* \
+    && yum update -y
 
-RUN ulimit -n 1024 \
+# We also need to disable mirrorlist and set a baseurl for the SCLo
+# repos we add here, and we need to do some extra work on arm to point
+# it at "altarch".
+RUN case "$HOSTTYPE" in \
+      aarch64) DIR=altarch ;; \
```

Comment:
You don't need to do the directory mapping explicitly, `archive.kernel.org/centos-vault` has the same structure as `mirror.centos.org`, so just replace the prefix, i.e. `mirror.centos.org` -> `archive.kernel.org/centos-vault`.  I tested this locally and it works.

Reply:
This mapping is necessary for the SCLo repos, because the original baseurl there is `mirror.centos.org/centos`, but that is wrong on aarch64 and we need it to be `archive.kernel.org/centos-vault/altarch`. The baseurls in the originally installed sources have `altarch` on arm, but not the new ones installed when we install centos-release-scl. That's why it wasn't working on ARM when we were doing the simple thing.

Reply:
ugh, ok.

Reply:
yeah.